android/src/main/java/com/reactnativeescposprinter/EscPosPrinterDiscoveryModule.java

```diff
@@ -216,32 +217,39 @@ public void onDiscovery(final DeviceInfo deviceInfo) {
         @Override
         public synchronized void run() {
           mPrinterList.add(deviceInfo);
-          WritableArray stringArray = Arguments.createArray();
-
-          for (int counter = 0; counter < mPrinterList.size(); counter++) {
-
-            final DeviceInfo info = mPrinterList.get(counter);
-            WritableMap printerData = Arguments.createMap();
-
-            String usbAddress = getUSBAddress(info.getTarget());
+        }
+      });
 
-            if (mExtractUsbSerialNumber && usbAddress != "") {
-              String usbSerialNumber = getUsbSerialNumber(usbAddress);
-              printerData.putString("usbSerialNumber", usbSerialNumber);
+      if (mOnDiscoveryReporter == null) {
+        mOnDiscoveryReporter = new Runnable() {
+          public synchronized void run() {
+            WritableArray stringArray = Arguments.createArray();
```

Comment:
let's extract this forming and sending data into separate method like: 
```
public void sendDiscoveredDataToJS() {

  WritableArray stringArray = Arguments.createArray();
  for (int counter = 0; counter < mPrinterList.size(); counter++) {
      final DeviceInfo info = mPrinterList.get(counter);
      WritableMap printerData = Arguments.createMap();
      
     ........
     
     sendEvent(reactContext, "onDiscoveryDone", stringArray);
     }
```

And try to send in right after discovery done (after 5 sec) here:

```
  private void performDiscovery(MyCallbackInterface callback) {  
    final Handler handler = new Handler();
    final Runnable r = new Runnable() { 
         public void run() {   
           if(mPrinterList.size() > 0) {
                sendDiscoveredDataToJS() // will be invoked after 5 sec with acc. results
           }
           stopDiscovery();       
           callback.onDone("Search completed"); 
           }   
        };
      handler.postDelayed(r, 5000);
     }
```